### PR TITLE
Add warning messages for using thread binding options on macOS

### LINKED ISSUE #4349: `--hpx:use-process-mask` option throw an exception on OS X
## Expected Behavior

I'm aware of the impossibility to do CPU binding on OS X, but usually requesting a binding on this kind of platform does not result in an error.

## Actual Behavior

The following exception is thrown
`hpx::init: hpx::exception caught: hwloc_get_cpubind failed: HPX(kernel_error)`

## Steps to Reproduce the Problem

Specifying the `--hpx:use-process-mask` option results in the exception.

No special code is needed to test it, even just the following one is enough:
```cpp
#include <hpx/hpx_init.hpp>

int hpx_main(int argc, char **argv) {
  return hpx::finalize();
}

int main(int argc, char **argv) {
  auto ret_code = hpx::init(hpx_main, argc, argv);

  return ret_code;
}
```

## Specifications

  - HPX Version: 1.4.0
  - Platform (compiler, OS): clang on OSX

### PR DESCRIPTION
**Pull request by msimberg**
Attempt to fix #4349. @albestro would you mind testing the changes?

### CODE REVIEW

File: src/util/command_line_handling.cpp [hunk 1]
```diff
@@ -816,6 +816,14 @@ namespace hpx { namespace util
                  vm.count("hpx:ignore-batch-env")) == 0) &&
             !use_process_mask_;
 
+#if defined(__APPLE__)
+        std::cerr
+            << "Warning: enabled process mask for thread binding, but "
+               "thread binding is not supported on macOS. Ignoring option."
+            << std::endl;
+        use_process_mask_ = false;
+#end
```
Comment (albestro) [inline src/util/command_line_handling.cpp]: Do you mean `#endif`?

Comment (albestro) [inline src/util/command_line_handling.cpp]:
```suggestion
#endif
```

File: src/util/command_line_handling.cpp [hunk 2]
```diff
@@ -816,6 +816,14 @@ namespace hpx { namespace util
                  vm.count("hpx:ignore-batch-env")) == 0) &&
             !use_process_mask_;
 
+#if defined(__APPLE__)
+        std::cerr
```
Comment (albestro) [inline src/util/command_line_handling.cpp]: Are you printing the warning message even if the `use_process_mask` option has not been used?

File: src/util/command_line_handling.cpp (hunk 1 above)
Comment (msimberg) [reply src/util/command_line_handling.cpp]: Excellent catch, thanks!

File: src/util/command_line_handling.cpp (hunk 2 above)
Comment (msimberg) [reply src/util/command_line_handling.cpp]: Also not intentional. Thanks!

**Comment by msimberg** [comment]
Thanks @albestro for testing! I've fixed the issues you found. Do the messages themselves seem clear enough for you?

**Comment by albestro** [comment]
Yeah, it looks perfect!
Thank you again!

**Pull request merged**